dsl_parser/import_resolver/abstract_import_resolver.py

```diff
@@ -42,12 +49,32 @@ def fetch_import(self, import_url):
 
 
 def read_import(import_url):
-    try:
-        with contextlib.closing(urllib2.urlopen(import_url)) as f:
-            return f.read()
-    except Exception, ex:
-        ex = exceptions.DSLParsingLogicException(
-            13, 'Import failed: Unable to open import url '
-            '{0}; {1}'.format(import_url, str(ex)))
-        ex.failed_import = import_url
-        raise ex
+    error_str = 'Import failed: Unable to open import url'
+    if import_url.startswith('file://'):
+        try:
+            with contextlib.closing(urllib2.urlopen(import_url)) as f:
+                return f.read()
+        except Exception, ex:
+            ex = exceptions.DSLParsingLogicException(
+                13, '{0} {1}; {2}'.format(error_str, import_url, ex))
+            raise ex
+    else:
+        num_retries = 0
+        while True:
+            try:
+                response = requests.get(import_url,
+                                        timeout=DEFAULT_REQUEST_TIMEOUT)
+            except Exception as err:
+                if num_retries >= DEFAULT_NUMBER_RETRIES:
+                    ex = exceptions.DSLParsingLogicException(
+                        13, '{0} {1}; {2}'.format(error_str, import_url, err))
+                    raise ex
+                time.sleep(DEFAULT_RETRY_DELAY)
+                num_retries += 1
+            else:
+                if 200 <= response.status_code < 300:
+                    return response.text
+                else:
+                    ex = exceptions.DSLParsingLogicException(
+                        13, '{0} {1};'.format(error_str, import_url))
```

Comment:
Now when I think of it, it will probably making sense adding the status code as part of the exception so the user isn't left in the blank